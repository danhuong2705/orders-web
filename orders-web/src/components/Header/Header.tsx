import React, { useContext } from "react";
import styled from "@emotion/styled";
import { Link } from "react-router-dom";
import CartIcon from "../../assets/cart.svg";
import LogoIcon from "../../assets/logo.svg";
import UserIcon from "../../assets/user.svg";
import { CartContext } from "../../contexts/cartContext";
const Wrapper = styled.div`
  height: 5rem;
  border-bottom: 1px solid #e0e0e0;
  position: fixed;
  top: 0;
  z-index: 1050;
  background: #ffffff;
  width: 100%;
  display: flex;
  justify-content: center;
`;
const Content = styled.div`
  width: 1264px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem 5rem;
`;

const Icon = styled.img`
  width: 1.5rem;
  height: 1.5rem;
  object-fit: contain;
`;
const Logo = styled(Icon)`
  width: 3rem;
  height: 3rem;
`;
const Badge = styled.div`
  background-color: red;
  color: #ffffff;
  font-size: 0.65rem;
  border-radius: 100%;
  position: absolute;
  bottom: 45%;
  width: 1rem;
  height: 1rem;
  display: flex;
  align-items: center;
  justify-content: center;
  left: 55%;
`;
const MenuItem = styled(Link)`
  margin: 0 1rem;
  position: relative;
`;
const RightMenuWrapper = styled.div`
  display: flex;
`;
const Header: React.FC = () => {
  const cartContext = useContext(CartContext);
  return (
    <Wrapper>
      <Content>
        <div>
          <Link to="/">
            <Logo src={LogoIcon} alt="logo" />
          </Link>
        </div>
        <RightMenuWrapper>
          <MenuItem to="/cart">
            <Icon src={CartIcon} alt="cart-icon" />
            {cartContext.totalProduct > 0 && (
              <Badge>{cartContext.totalProduct}</Badge>
            )}
          </MenuItem>
          <MenuItem to="/order-list">
            <Icon src={UserIcon} alt="user-icon" />
          </MenuItem>
        </RightMenuWrapper>
      </Content>
    </Wrapper>
  );
};

export default Header;
