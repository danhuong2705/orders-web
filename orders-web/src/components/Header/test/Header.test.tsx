import React from "react";
import Header from "..";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe("<Header />", () => {
  it("should render correctly", async () => {
    const { container } = await render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
