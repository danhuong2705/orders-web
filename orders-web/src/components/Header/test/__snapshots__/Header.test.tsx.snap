// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<div
  class="css-vlq267"
>
  <div
    class="css-1aya70s"
  >
    <div>
      <a
        href="/"
      >
        <img
          alt="logo"
          class="css-13dskwy"
          src="logo.svg"
        />
      </a>
    </div>
    <div
      class="css-ov1ktg"
    >
      <a
        class="css-15tgtd"
        href="/cart"
      >
        <img
          alt="cart-icon"
          class="css-1acylx9"
          src="cart.svg"
        />
      </a>
      <a
        class="css-15tgtd"
        href="/order-list"
      >
        <img
          alt="user-icon"
          class="css-1acylx9"
          src="user.svg"
        />
      </a>
    </div>
  </div>
</div>
`;
